packages/components/src/components/multi-select/multi-select/multi-select-styles.ts

```diff
@@ -40,209 +42,207 @@ export const getComponentCss = (
     'no-results': {
       padding: `${spacingStaticSmall} 12px`,
+      boxSizing: 'border-box',
       color: contrastHighColor,
       ...prefersColorSchemeDarkMediaQuery(theme, {
         color: contrastHighColorDark,
       }),
-      boxSizing: 'border-box',
       ...getNoResultsOptionJssStyle(),
     },
-    ...getFunctionalComponentRequiredStyles(),
+    // TODO: maybe we should extract it as functional component too
+    'sr-only': getHiddenTextJssStyle(),
+    // .label / .required
+    ...getFunctionalComponentLabelStyles(isDisabled, hideLabel, theme),
+    // .message
     ...getFunctionalComponentStateMessageStyles(theme, state),
-    'sr-text': getHiddenTextJssStyle(),
   });
 };
 
-const getInputStyles = (isDisabled: boolean, theme: Theme): Styles => {
-  const { primaryColor, disabledColor } = getThemedColors(theme);
-  const { primaryColor: primaryColorDark, disabledColor: disabledColorDark } = getThemedColors('dark');
+// TODO: use getSlottedTextFieldTextareaSelectStyles() instead an manipulate selectors like done with PIN Code or even better make it configurable as parameter
+const getInputStyles = (
+  isDisabled: boolean,
+  direction: SelectDropdownDirectionInternal,
+  isOpen: boolean,
+  state: FormState,
+  theme: Theme
+): JssStyle => {
+  const isDirectionDown = direction === 'down';
+  const { primaryColor, disabledColor, backgroundColor, contrastMediumColor } = getThemedColors(theme);
+  const {
+    primaryColor: primaryColorDark,
+    disabledColor: disabledColorDark,
+    backgroundColor: backgroundColorDark,
+    contrastMediumColor: contrastMediumColorDark,
+  } = getThemedColors('dark');
+  const { formStateColor, formStateHoverColor } = getThemedFormStateColors(theme, state);
+  const { formStateColor: formStateColorDark, formStateHoverColor: formStateHoverColorDark } = getThemedFormStateColors(
+    'dark',
+    state
+  );
 
   return {
-    input: {
-      flex: 1,
-      minWidth: 0,
-      height: `calc(${INPUT_HEIGHT_CALC})`, // we need 6px additionally so input height becomes 50px
-      font: textSmallStyle.font.replace('ex', 'ex + 6px'), // a minimum line-height is needed for input, otherwise value is scrollable in Chrome, +6px is alig
-      padding: `${inputYPadding} ${spacingStaticMedium}`,
-      boxSizing: 'border-box',
-      border: 0, // done via container
-      outline: 0,
-      appearance: 'none',
-      background: 'transparent',
-      ...textSmallStyle,
-      textOverflow: 'ellipsis',
-      '&:disabled': {
-        cursor: 'not-allowed',
+    gridArea: '1/1/1/-1',
+    flex: 1,
+    minWidth: 0,
+    // TODO: abstract and re-use for multi-select, select-wrapper and text-field-wrapper
+    height: `calc(${fontLineHeight} + 10px + ${borderWidthBase} * 2 + ${spacingStaticSmall} * 2)`, // we need 10px additionally so input height becomes 54px, // we need 6px additionally so input height becomes 50px
```

Comment:
Are these comments correct? Also check comment on next line

Reply:
Yeah, because most of those styles should/might be the same.